src/zzz_od/gui/view/home/home_interface.py

```
@@ -844,22 +846,66 @@ def _extract_color_from_image(self) -> tuple[int, int, int]:
+        if w <= 0 or h <= 0:
+            return 64, 158, 255
+
+        # 计算采样步长，使得样本点约为 100x100 或更少
+        target_samples = 10000  # 上限采样点数
+        step_x = max(1, int(w / min(w, max(1, int((target_samples ** 0.5)) ))))
+        step_y = max(1, int(h / min(h, max(1, int((target_samples ** 0.5)) ))))
+
+        hue_acc = 0.0
+        weight_acc = 0.0
+
+        # 遍历采样点，累积以 saturation*value 作为权重，偏重鲜艳明亮的颜色
```

Ruff (0.14.3)

Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)